### PR TITLE
cilium: further improve local address selection

### PR DESCRIPTION
**Pull request by borkmann**
(See commit desc.)

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/8585)

### CODE REVIEW

**Comment by borkmann** [comment]
test-me-please

File: pkg/ip/ip.go [hunk 1]
```diff
@@ -737,7 +740,7 @@ func KeepUniqueIPs(ips []net.IP) []net.IP {
 
 var privateIPBlocks []*net.IPNet
 
-func init() {
+func initPrefixes() {
```
Comment (brb) [inline pkg/ip/ip.go]: Nit: `s/initPrefixes/initPrivateIPBlocks`?

File: pkg/ip/ip.go [hunk 2]
```diff
@@ -750,6 +753,57 @@ func init() {
...
+var excludedIPs []net.IP
+
+func initExcludedIPs() {
+	// We exclude below bad device prefixes from address selection ...
+	prefixes := []string{
+		"docker",
+	}
+	links, err := netlink.LinkList()
+	if err != nil {
+		return
```
Comment (brb) [inline pkg/ip/ip.go:765]: Maybe makes sense to log an error here?

**Review by brb — APPROVED** [review]
LGTM! Two nits which can be ignored.

**Comment by borkmann** [comment]
test-me-please

File: pkg/ip/ip.go (hunk 2 above)
Comment (borkmann) [reply pkg/ip/ip.go:765]: Probably true, but then I guess we should log also the many other places for similar calls which we don't do right now. :/

**Pull request merged**